osu.Game/Screens/Play/Player.cs

```diff
@@ -800,7 +768,13 @@ private void progressToResults(bool withDelay)
 
             resultsDisplayDelegate = new ScheduledDelegate(() =>
             {
-                if (prepareScoreForDisplayTask?.IsCompleted != true)
+                if (prepareScoreForDisplayTask == null)
```

Comment:
Isn't this going to run multiple times? This delegate is looping.

Reply:
Potentially yes in cases where the the `beginScoreImport` returns on the if check. I don't necessarily see it as an issue because it is essentially behaving as before, and the `Scheduler` might stop working soon once the screen exits (not sure about this behavior). If desired for the `canShowResults ` check can be used at the beginning of `progressToResults` as well.

Reply:
I dunno, I can't follow this logic. Like why is this only run when the task is null? There's too much missing detail which makes it hard to follow the changes.